ec/src/models/double_odd/group.rs

```diff
@@ -0,0 +1,461 @@
+            P::BaseField::ZERO,
+            P::BaseField::ZERO,
+        )
+    }
+
+    #[inline]
+    fn is_zero(&self) -> bool {
+        (self.e == self.z) && self.u == P::BaseField::from(0)
+    }
+}
+
+impl<P: DOCurveConfig> AdditiveGroup for Projective<P> {
+    type Scalar = P::ScalarField;
+
+    const ZERO: Self = Self::new_unchecked(
+        P::BaseField::ONE,
+        P::BaseField::ONE,
+        P::BaseField::ZERO,
+        P::BaseField::ZERO,
+    );
+
+    // Implements extended coordinates doubling as per the DO website:
+    // https://doubleodd.group/formulas-eu.html#extended-coordinates
+    // https://web.archive.org/web/20240718235643/https://doubleodd.group/formulas-eu.html#extended-coordinates
+    fn double_in_place(&mut self) -> &mut Self {
+        self.z = -P::get_c().double() * self.t.square(); // Self.z == -2cT^2
+        self.t = self.e;
+        self.e.square_in_place(); // Self.e == E^2
+        self.z += self.e; // Self.z == E^2 -2cT^2
+        self.z += (P::COEFF_A * self.u.square()).double(); // Self.z == W' == E^2 + 2aU^2 -2cT^2
+
+        self.t *= self.u;
+        self.t.double_in_place(); // Self.t == J'= 2EU
+
+        self.u = self.t; // Self.u == Self.t == J'
+        self.t.square_in_place(); // Self.t == T' = J'^2
+
+        self.u *= self.z; // U' = J' * W'
+        self.z.square_in_place(); // Self.z == W'^2
+
+        self.e.square_in_place(); // Self.e == X' == E^4
+        self.e.double_in_place(); // Self.e == 2X'
+        self.e += -self.z + (P::COEFF_A * self.t); // E' = 2X' - Z' + aT'
+
+        self
+    }
+}
+
+impl<P: DOCurveConfig> PrimeGroup for Projective<P> {
+    type ScalarField = P::ScalarField;
+
+    #[inline]
+    fn generator() -> Self {
+        Affine::generator().into()
+    }
+
+    #[inline]
+    fn mul_bigint(&self, other: impl AsRef<[u64]>) -> Self {
+        P::mul_projective(self, other.as_ref())
+    }
+}
+
+impl<P: DOCurveConfig> CurveGroup for Projective<P> {
+    type Config = P;
+    type BaseField = P::BaseField;
+    type Affine = Affine<P>;
+    type FullGroup = Affine<P>;
+
+    #[inline]
+    fn normalize_batch(v: &[Self]) -> Vec<Self::Affine> {
+        let mut z_s = v.iter().map(|g| g.z).collect::<Vec<_>>();
+        ark_ff::batch_inversion(&mut z_s);
+
+        // Perform affine transformations
+        ark_std::cfg_iter!(v)
+            .zip(z_s)
+            .map(|(g, z)| match g.is_zero() {
+                true => Affine::identity(),
+                false => {
+                    let e = g.e * z;
+                    let u = g.u * z;
+
+                    Affine::new_unchecked(e, u)
+                },
+            })
+            .collect()
+    }
+}
+
+impl<P: DOCurveConfig> Neg for Projective<P> {
+    type Output = Self;
+
+    #[inline]
+    fn neg(mut self) -> Self {
+        self.u = -self.u;
+        self
+    }
+}
+
+impl<P: DOCurveConfig, T: Borrow<Affine<P>>> AddAssign<T> for Projective<P> {
+    fn add_assign(&mut self, other: T) {
```

Comment:
Can you add a reference for this code? Perhaps a pointer to existing "projective + projective" formulae, along with a description of the simplification when one of the points is affine.